snuba/processor.py

```diff
@@ -634,3 +634,22 @@ def process_delete_tag(self, message):
         query_time_flags = (NEEDS_FINAL, message['project_id'])
 
         return (count_query_template, insert_query_template, query_args, query_time_flags)
+
+class SpansProcessor(Processor):
+    """
+    A processor for session tracking spans.
+    """
+    def __init__(self, SCHEMA):
+        super(SpansProcessor, self).__init__(SCHEMA)
+
+    def process_insert(self, message):
+        # simply make sure all the keys we want exist
+        if not all(col.flattened in message for col in self.SCHEMA.ALL_COLUMNS):
+            return None
+
+        message['timestamp'] = _ensure_valid_date(datetime.strptime(message['timestamp'], settings.PAYLOAD_DATETIME_FORMAT))
```

Comment:
Should any additional columns be removed? Or will a lower layer remove/ignore unwanted columns?

Reply:
`row_from_processed_event` takes only the values that are in the schema, so everything else is thrown out.

FWIW , I don't think this process_insert function will survive contact with reality. It's nice to assume that all the values you need are all there and correctly formatted, but I'm gonna bet we have to start adding exceptions real soon.